svtyper/parsers.py

```diff
@@ -0,0 +1,1254 @@
+                a = line[line.find('<')+1:line.find('>')]
+                r = re.compile(r'(?:[^,\"]|\"[^\"]*\")+')
+                self.add_info(*[b.split('=')[1] for b in r.findall(a)])
+            elif line.split('=')[0] == '##ALT':
+                a = line[line.find('<')+1:line.find('>')]
+                r = re.compile(r'(?:[^,\"]|\"[^\"]*\")+')
+                self.add_alt(*[b.split('=')[1] for b in r.findall(a)])
+            elif line.split('=')[0] == '##FORMAT':
+                a = line[line.find('<')+1:line.find('>')]
+                r = re.compile(r'(?:[^,\"]|\"[^\"]*\")+')
+                self.add_format(*[b.split('=')[1] for b in r.findall(a)])
+            elif line[0] == '#' and line[1] != '#':
+                self.sample_list = line.rstrip().split('\t')[9:]
+            elif line.startswith('##fileDate='):
+                pass
+            else:
+                self.header_misc.append(line.rstrip())
+
+    # return the VCF header
+    def get_header(self):
+        header = '\n'.join(['##fileformat=' + self.file_format,
+                            '##fileDate=' + time.strftime('%Y%m%d'),
+                            '##reference=' + self.reference] + \
+                           [i.hstring for i in self.info_list] + \
+                           [a.hstring for a in self.alt_list] + \
+                           [f.hstring for f in self.format_list] + \
+                           self.header_misc + \
+                           ['\t'.join([
+                               '#CHROM',
+                               'POS',
+                               'ID',
+                               'REF',
+                               'ALT',
+                               'QUAL',
+                               'FILTER',
+                               'INFO',
+                               'FORMAT'] + \
+                                      self.sample_list
+                                  )])
+        return header
+
+    def write_header(self, fd=None):
+        if fd is None:
+            fd = sys.stdout
+        print(self.get_header(), file=fd)
+
+    def add_info(self, id, number, type, desc):
+        if id not in [i.id for i in self.info_list]:
+            inf = self.Info(id, number, type, desc)
+            self.info_list.append(inf)
+
+    def add_alt(self, id, desc):
+        if id not in [a.id for a in self.alt_list]:
+            alt = self.Alt(id, desc)
+            self.alt_list.append(alt)
+
+    def add_format(self, id, number, type, desc):
+        if id not in [f.id for f in self.format_list]:
+            fmt = self.Format(id, number, type, desc)
+            self.format_list.append(fmt)
+
+    def add_sample(self, name):
+        self.sample_list.append(name)
+
+    # get the VCF column index of a sample
+    # NOTE: this is zero-based, like python arrays
+    def sample_to_col(self, sample):
+        return self.sample_list.index(sample) + 9
+
+    def _init_bnd_breakpoint_func(self):
```

Comment:
It doesn't look to me like this uses the object. Perhaps this should be changed to `@staticmethod`